### DISCUSSION TITLE
Decrease a virtual axis overtime

### DISCUSSION THREAD

**Discussion by Svennixx** [WhiteMagic/JoystickGremlin#507] [Q&A] [answered]
I'm trying to decrease the virtual X axis gently from a set point to 0. Not an instant 100 -> 0 but over a span of 600ms. 
Is this possible? I tried to use a simple time.sleep function but that halts the whole plugin/Gremlin for that time I set to wait.

**Comment by WhiteMagic** [comment, marked as answer]
You can use `time.sleep()`, however, this needs to happen in a separate thread as you otherwise sleep the main thread. You could technically also utilize a repeating macro to get a similar effect though depending on the resolution you're after it might be a bit ugly.

**Reply by Svennixx** [reply]
Ah of course! Why didn't I think of that, thank you works great now
